### ISSUE TITLE
[BUG] [python-flask] invalid `.from_dict` generated when using object as query parameter

### ISSUE THREAD

**Issue by Tohnmeister** [OpenAPITools/openapi-generator#4873]
#### Bug Report Checklist

- [X] Have you provided a full/minimal spec to reproduce the issue?
- [X] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [X] What's the version of OpenAPI Generator used?
4.2.2
- [X] Have you search for related issues/PRs?
- [X] What's the actual output vs expected output?
I would've expected this to generate code that compiles/runs. But the `.from_dict` seems invalid Python syntax.
- [ ] [Optional] Bounty to sponsor the fix ([example](https://www.bountysource.com/issues/66123212-javascript-client-produces-a-wrong-object-for-a-string-enum-type-that-is-used-with-ref))

##### Description

When using this spec as input:

```yaml
openapi: "3.0.0"
info:
  title: "My API"
  version: "0.0.1"
paths:
  /invoices:
    get:
      parameters:
        - name: filter
          in: query
          description: Filter parameters
          required: false
          schema:
            type: object
            properties:
              field:
                type: string
      responses:
        '200':
          description: An array of invoices
          content:
            application/json:
              schema:
                type: array
                items:
                  type: object
                  properties:
                    number:
                      type: integer
                      format: int64
```

a default_controller.py is generated containing the following code:

```python
import connexion
import six

from openapi_server.models.filter import Filter  # noqa: E501
from openapi_server.models.inline_response200 import InlineResponse200  # noqa: E501
from openapi_server import util


def invoices_get(filter=None):  # noqa: E501
    """invoices_get

     # noqa: E501

    :param filter: Filter parameters
    :type filter: dict | bytes

    :rtype: List[InlineResponse200]
    """
    if connexion.request.is_json:
        filter = .from_dict(connexion.request.get_json())  # noqa: E501
    return 'do some magic!'
```
The `.from_dict` is invalid Python syntax.

Trying to run the generated application, results in the following error:

```bash
jsonschema.exceptions.RefResolutionError: Unresolvable JSON pointer: 'components/parameters/filter'
```

##### openapi-generator version

4.2.2

##### OpenAPI declaration file content or url

See above.

##### Command line used for generation

openapi-generator generate -i ./openapi.yaml -g python-flask -o ./generated

##### Steps to reproduce

- Use above command and openapi.yaml to generate python-flask code.

##### Suggest a fix

I suppose the .from_dict needs to be prefixed by a class name.

**Comment by LongBeachHXC** [comment]
Is there a fix for this yet? I am experiencing this issue when trying to generate python server config using aiohttp

**Comment by spacether** [comment]
@Tohnmeister I think that this is a problem with your spec, not the generator. 
Have toy tried sending a string query parameter?
I don't think type object is allowed as a query parameter.

**Comment by spacether** [comment]
My mistake, we can send objects per: https://swagger.io/docs/specification/describing-parameters/
```
Query parameters can be primitive values, arrays and objects. OpenAPI 3.0 provides several ways to serialize objects and arrays in the query string.

Arrays can be serialized as:

form – /products?color=blue,green,red or /products?color=blue&color=green, depending on the explode keyword
spaceDelimited (same as collectionFormat: ssv in OpenAPI 2.0) – /products?color=blue%20green%20red
pipeDelimited (same as collectionFormat: pipes in OpenAPI 2.0) – /products?color=blue|green|red
Objects can be serialized as:
form – /points?color=R,100,G,200,B,150 or /points?R=100&G=200&B=150, depending on the explode keyword
deepObject – /points?color[R]=100&color[G]=200&color[B]=150
```

**Comment by JustinBusschau** [comment]
Correct, @spacether 

Also if I define an enum as:
```yaml
openapi: 3.0.0
components:
  schemas:
    rateEnum:
      enum:
        - off
        - low
        - economy
        - medium
        - high
      type: string
```

and then pass that as an input parameter to a `PATCH` call, that should serialise as something like:
```http
PATCH /api/consumer?rate=economy
```

That all works fine, but there are no samples that do this in the `petstore.yaml` so I think this is why the failure to generate a valid handler/controller for this in the flask servers was never identified and there is no way of testing this using the petstore sample

**Comment by praveen049** [comment]
Hi
   I am having similar problem when using object in query string. The code generated has 
`.from_dict(connexion.request.get_json())`

Is there any fix or workaround for this issue  ?

Thanks

**Comment by Tohnmeister** [comment]
> Hi
> I am having similar problem when using object in query string. The code generated has
> `.from_dict(connexion.request.get_json())`
> 
> Is there any fix or workaround for this issue ?
> 
> Thanks

If I read the previous comments correctly there was a [PR](https://github.com/OpenAPITools/openapi-generator/pull/5372) which fixes this issue. It was included in version 4.3.0.

Haven't tested that yet though. I worked around the issue by not using an object in the query anymore. So not really a workaround. :-)

**Comment by srl295** [comment]
> > Hi
> > I am having similar problem when using object in query string. The code generated has
> > `.from_dict(connexion.request.get_json())`
> > Is there any fix or workaround for this issue ?
> > Thanks
> 
> If I read the previous comments correctly there was a [PR](https://github.com/OpenAPITools/openapi-generator/pull/5372) which fixes this issue. It was included in version 4.3.0.
> 
> Haven't tested that yet though. I worked around the issue by not using an object in the query anymore. So not really a workaround. :-)

yeah, i switched to openapi-generator because of https://github.com/swagger-api/swagger-codegen/issues/10521 - I have my analysis there, seems like the same issue. and does seem fixed.

**Comment by spacether** [comment]
Closing this due to reports of it being fixed in 4.3.0 and higher
